src/commcare_cloud/commands/ansible/service.py

```diff
@@ -1,328 +1,540 @@
+
+    def execute_action(self, action, host_pattern=None, process_pattern=None):
+        if action == 'status':
+            return self.check_status(host_pattern, process_pattern)
+
+        def _change_state(service_name, run_on, action=action):
+            service_args = 'name={} state={}'.format(service_name, STATES[action])
+            return self._run_ansible_module(run_on, 'service', service_args)
+
+        return self._run_action_on_hosts(_change_state, host_pattern, process_pattern)
+
+    def _run_action_on_hosts(self, action_fn, host_pattern, process_pattern):
+        if host_pattern:
+            self.environment.inventory_manager.subset(host_pattern)
+
+        non_zero_exits = []
+        ran = False
+        for service, run_on in self._get_service_host_groups(process_pattern):
+            hosts = self.environment.inventory_manager.get_hosts(run_on)
+            run_on = ','.join([host.name for host in hosts]) if host_pattern else run_on
+            if hosts:
+                ran = True
+                exit_code = action_fn(service, run_on)
+                if exit_code != 0:
+                    non_zero_exits.append(exit_code)
+
+        if not ran:
+            raise NoHostsMatch
+        return non_zero_exits[0] if non_zero_exits else 0
+
+    def _get_service_host_groups(self, process_pattern):
+        service_groups = []
+        if process_pattern:
+            for name in process_pattern.split(','):
+                service_groups.append(self.service_process_mapping[name])
         else:
-            return self.SERVICES[service_group]
+            service_groups = list(self.service_process_mapping.values())
+        return service_groups
+
+
+class Nginx(AnsibleService):
+    name = 'nginx'
+    inventory_groups = ['proxy']
+
+
+class Elasticsearch(AnsibleService):
+    name = 'elasticsearch'
+    inventory_groups = ['elasticsearch']
+
 
-    def run_for_es(self, args, unknown_args):
-        action = args.action
-        if action == "restart":
-            return RestartElasticsearch(self.parser).run(args, unknown_args)
+class Couchdb(AnsibleService):
+    name = 'couchdb'
+    inventory_groups = ['couchdb2']
+
+
+class RabbitMq(AnsibleService):
+    name = 'rabbitmq'
+    inventory_groups = ['rabbitmq']
+    service_name = 'rabbitmq-server'
+
+
+class Redis(AnsibleService):
+    name = 'redis'
+    inventory_groups = ['redis']
+    service_name = 'redis-server'
+
+
+class Riakcs(MultiAnsibleService):
+    name = 'riakcs'
+    service_process_mapping = {
+        'riak': ('riak', 'riakcs'),
+        'riakcs': ('riak-cs', 'riakcs'),
+        'stanchion': ('stanchion', 'stanchion'),
+    }
+
+
+class Kafka(MultiAnsibleService):
+    name = 'kafka'
+    service_process_mapping = {
+        'kafka': ('kafka-server', 'kafka'),
+        'zookeeper': ('zookeeper', 'zookeeper')
+    }
+
+
+class Postgres(MultiAnsibleService):
+    name = 'postgres'
```

Comment:
Should this say 'postgresql' to match the inventory host name? Or did you change this intentionally?

Reply:
I did change this intentionally since it's a few less characters to type and since it also isn't actually postgresql anymore but a combined service. Fine to change back though if you don't think it's a good change